### ISSUE TITLE
Custom Linear relaxations for feasibility pump

### ISSUE THREAD

**Issue by this-josh** [lanl-ansi/Juniper.jl#254]
I have a non-linear program
```Julia
juniper_opt = optimizer_with_attributes(Juniper.Optimizer, "nl_solver" => Ipopt.Optimizer, "mip_solver" => Gurobi.Optimizer)
model = Model(juniper_opt)
@variable(model, a)
@NLconstraint(model, a * abs(a) >=3)
@objective(model, Min, a)
optimize!(model)
```
Now when Juniper creates a MILP to do the feasibility pump it's my understanding it uses everything but the non-linear constraints. In my instance I have a linear relaxation of the non-linear constraints. So I've made it so that Juniper can accept an `mip` model for the feasibility pump.
```Julia
mip = Model(juniper_opt)
@variable(mip, a)
@constraint(mip, a * a >=3)
@objective(mip, Min, a)
set_optimizer_attribute(model, "mip_model", mip)

optimize!(model)
```
Essentially all that changes is in the [fpump code](https://github.com/lanl-ansi/Juniper.jl/blob/2ab639e80e73102b992540eb4146882ddb628c51/src/fpump.jl#L353) I pass the mip model, perhaps there's more that could be done. For my use case this is helpful as my linear relaxation won't remove any valid solutions and it can quickly show a lot to be infeasible. 

Is this code of interest? If so, I can open a PR.

**Comment by Wikunia** [comment]
Yes that would be a nice addition. It currently removes all non linear constraints and changes the objective such that it minimizes the difference to the NLP relaxation. 
Looking forward to your PR 😊

**Comment by this-josh** [comment]
After reviewing my current implementation I think I need to better understand MOI before I can do a good job of this. For now I'll have to put this on pause, but it is still on my lists of things to do.
